clang/runtime/dpct-rt/include/util.hpp

```diff
@@ -286,6 +286,73 @@ T permute_sub_group_by_xor(sycl::sub_group g, T x, unsigned int mask,
                                      : id);
 }
 
+/// match_any_over_sub_group return a mask in which some bits are set to 1,
+/// indicating that the values provided by the work-item represented by
+/// these bits are equal. The n-th bit of mask representing the work-item
+/// with id n. The parameter \p member_mask indicating the work-items
+/// participating the call.
+/// \tparam T Input value type
+/// \param [in] g Input sub_group
+/// \param [in] member_mask Input mask
+/// \param [in] value Input value
+/// \returns The result
+template <typename T>
+unsigned int match_any_over_sub_group(sycl::sub_group g, unsigned member_mask,
+                                      T value) {
+  if (!member_mask) {
+    return 0;
+  }
+  unsigned int id = g.get_local_linear_id();
+  unsigned int flag = 0, result = 0, reduce_result = 0;
+  unsigned int bit_index = 0x1 << id;
+  bool is_participate = member_mask & bit_index;
+  T broadcast_value = 0;
```

Comment:
Add static assert `std::is_arithmetic`, or this line may cause unexpected error.